src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -217,6 +217,21 @@ public async Task Connect_WithoutListener_ThrowSocketExceptionWithAppropriateInf
                 Assert.Contains(a.LocalEndPoint.ToString(), ex.Message);
             }
         }
+
+        [Theory]
+        [MemberData(nameof(LoopbacksAndAny))]
+        public async Task Connect_DatagramSockets_DontThrowConnectedException_OnSecondAttempt(IPAddress listenAt, IPAddress secondConnection)
+        {
+            using Socket listener = new Socket(listenAt.AddressFamily, SocketType.Dgram, ProtocolType.Udp);
+            using Socket s = new Socket(listenAt.AddressFamily, SocketType.Dgram, ProtocolType.Udp);
+            listener.Bind(new IPEndPoint(listenAt, 0));
+
+            await ConnectAsync(s, new IPEndPoint(listenAt, ((IPEndPoint)listener.LocalEndPoint).Port));
+            Assert.True(s.Connected);
+
+            await ConnectAsync(s, new IPEndPoint(secondConnection, PlatformDetection.IsOSX ? 1 : 0));
+            Assert.True(s.Connected);
```

Comment:
What are the arguments for keeping `Connected == true` here? What will `getpeername` return for a fresh socket VS a connected socket VS a socket disconnected with `Connect(IPAddress.Any, 0)`

Reply:
> What are the arguments for keeping `Connected == true` here?

I observed the `.NET Framework`'s behavior here.

> What will `getpeername` return for a fresh socket VS a connected socket VS a socket disconnected with `Connect(IPAddress.Any, 0)`?

I need to check this out

Reply:
> I observed the .NET Framework's behavior here.

Then let's keep that behavior for compatibility 👍